src/Common/Config/ConfigProcessor.cpp

```diff
@@ -535,6 +544,9 @@ void ConfigProcessor::doIncludesRecursive(
 
     if (attr_nodes["from_env"]) /// we have env subst
     {
+        if (node->hasChildNodes()) /// only allow substitution for nodes with no value
+            throw Poco::Exception("Element <" + node->nodeName() + "> has value, can't process from_env substitution");
```

Comment:
I used this behavior before (define default value in the xml and override it via env) for development, AFAICS the bug that described in https://github.com/ClickHouse/ClickHouse/issues/56537 is fixed by removing `SUBSTITUTION_ATTRS` from source node during merging before?